proposals/fate-operator-proposal.md

```diff
@@ -0,0 +1,220 @@
+3. [A case of traffic violations insurance-using federated learning](https://www.fedai.org/cases/a-case-of-traffic-violations-insurance-using-federated-learning/)
+4. [Utilization of FATE in Anti Money Laundering Through Multiple Banks](https://www.fedai.org/cases/utilization-of-fate-in-anti-money-laundering-through-multiple-banks/)
+
+Other Federated Machine Learning cases:
+1. [Federated Learning for Mobile Keyboard Prediction](https://research.google/pubs/pub47586/)
+2. [Federated Learning powered by NVIDIA Clara](https://devblogs.nvidia.com/federated-learning-clara/): hospitals and medical institutions collaboratively share and combine their local knowledge
+3. [Owkin Launches the Collaborative COVID-19 Open AI Consortium (COAI)](https://www.unite.ai/covid-19-open-ai-consortium/)
+
+## Motivation
+Kubeflow provides a toolset for end-to-end machine learning workflow on Kubernetes. Introducing the capability of federated learning to Kubeflow helps FML users and researchers leverage existing Kubeflow toolkits in their workflows and help them more efficiently build FML solutions. 
+
+A FATE-Operator is a start of supporting FML in Kubeflow. This proposal is aimed to defining what FATE operator should look like, and how to apply to Kubeflow.
+
+## Goals
+A Kubeflow user should be able to run training using FATE as easily as they can using PyTorch, Tensorflow. This proposal is centered around a Kubernetes Operator for FATE. With the FATE-Operator, a user can:
+1.	Provision and manage a FATE cluster;
+2.	Submit an FML job to FATE.
+
+This proposal defines the following:
+1.	A FATE operator with three CRDs: 
+   * FateJob: create an FML job;
+   * FateCluster: create a FATE cluster to serve FML jobs;
+   * Kubefate: the resource management component of FATE cluster.
+2.	Example of full lifecycle to create KubeFATE component, deploy FATE cluster and submit an FML job to created FATE and get the result. Note that, KubeFATE and FATE cluster only needs to be deployed once, and can handle multiple jobs.
+
+## Non-Goals
+For the scope of this proposal, we won’t be addressing the method of serving the model.
+
+## Design
+
+### Container images and deploying FATE cluster on Kubernetes
+We have built a set of Docker images for FATE cluster, and put into: https://hub.docker.com/orgs/federatedai/repositories . All the images have been already used and verified by community users. 
+
+There is a provisioning and management component of FATE cluster, called [KubeFATE](https://github.com/FederatedAI/KubeFATE/tree/master/k8s-deploy). KubeFATE manages FATE clusters of one party in a federation.
+
+All images work well and are proven in users’ environments.
+
+### Custom Resource Definition
+#### Kubefate
+```
+apiVersion: app.kubefate.net/v1beta1
+kind: Kubefate
+metadata:
+  name: kubefate-sample
+  namespace: kube-fate
+spec:
+  # kubefate image tag
+  imageVersion: v1.0.2
+  # ingress host
+  ingressDomain: kubefate.net
```

Comment:
Is Kubefate creating the ingress and setting up networking? The architecture in Kubeflow is to use ISTIO to create a service mesh. Different platforms can then expose the external endpoint as makes sense on their platform.

Reply:
There are three ingress will exposed: 1. The KubeFATE service will exposed for serving management requests; 2. The FATE-Board component of FATE; 3. The Proxy to connect with other organizations. The first two have been changed to istio gateway. The proxy now is expose with nodeport and connect with other organization. We are working on refactor it to expose gateway now. It will take some times.